Boost robustness

jessetilro opened the pull request:
Just preparing the PR and checking build status. Not ready yet, more commits are coming!

Commit:
Add abstract super class for activities.

Added an abstract super class for the activities, to share some common
activity logic and treat them uniformly. Also updated some integrations.
This is all the foundation for further improvements on the networking and
app stability which will be done in upcoming commits.

Commit:
Merge branch 'release' into boost-robustness

Commit:
Update on disconnect go to main.

When for whatever reason the client gets disconnected from the server, the
application immediately resets and returns to the main activity.

Commit:
Fix starting an stopping the game.

Fixed starting and stopping the game to work better and cover more cases.
Refactor to not use static methods.

Commit:
Update style and Fix connection issues.

Major updates to the styling. Put them together with bug fixes in one
large commit, because I figured why the hell not.

The game now has a custom splash screen. Updated color scheme, still have
some work to do (background colors in different activities). Manual
refresh button in lobby (not really necessary since everything should work
automatically, but just in case.

Fixed connection issues some more.

Commit:
Tweak the layout of some views related to Gunner.

Update layout consistency and made it shine a bit more for the views
related to the Gunner role. Will do the same for the remaining activities.

Commit:
Update to support server Crew List in GUI.

Commit:
Tweak layout.

Updated remaining activities to have theme consistent layouts. The
location activities now also display the role they are for.

dmalarev commented:
Lgtm

wouterz commented on app/src/main/java/nl/tudelft/pixelperfect/pixelperfect/location/LocationArmoryActivity.java:
```diff
@@ -28,19 +30,35 @@
+            case android.R.id.home:
+                onBackPressed();
+                return true;
+        }
```
switch with one case? feel free to leave it but not that pretty

jessetilro replied:
This was already done this way in other classes, just copy pasted it.

wouterz commented on app/src/main/java/nl/tudelft/pixelperfect/pixelperfect/location/LocationEngineroomActivity.java:
```diff
@@ -29,19 +31,35 @@
+        switch (item.getItemId()) {
+            case android.R.id.home:
+                onBackPressed();
+                return true;
```
again switch with one case

jessetilro replied:
This was already done this way in other classes, just copy pasted it.

wouterz commented on app/src/main/java/nl/tudelft/pixelperfect/pixelperfect/location/LocationLabActivity.java:
```diff
@@ -27,19 +29,35 @@
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case android.R.id.home:
+                onBackPressed();
```
same

jessetilro replied:
This was already done this way in other classes, just copy pasted it.

wouterz commented:
dammn jesse super netjes weer!

Commit:
Fix Plasma Leak Minigame Layout.

This layout was not understood well by nearly all players during the
usability experiment, so I cleaned it up a bit.

jessetilro commented:
Completed last fix just a minute ago. Ready to merge if Travis agrees.
